inc/admin.php

```diff
@@ -22,6 +26,32 @@
 function bootstrap() : void {
 	// Actions.
 	add_action( 'admin_init', __NAMESPACE__ . '\\init_admin_cols', 99 );
+	add_action( 'user_profile_update_errors', __NAMESPACE__ . '\\remove_required_fields_errors', 10, 3 );
+}
+
+/**
+ * Removes required fields errors from the user profile update.
+ *
+ * @param WP_Error $errors WP_Error object (passed by reference).
+ * @param bool     $update Whether this is a user update.
+ * @param stdClass $user   User object (passed by reference).
+ * @return void
+ */
+function remove_required_fields_errors( $errors, $update, $user ) : void {
+	if ( $user->role !== GUEST_ROLE ) {
+		return;
+	}
+
+	$error_codes = $errors->get_error_codes();
+	$error_codes_to_remove = [ 'empty_email', 'nickname' ];
+	$current_codes = array_intersect( $error_codes, $error_codes_to_remove );
+	if ( count( $current_codes ) === 0 ) {
+		return;
+	}
```

Comment:
This check is probably redundant and can be removed. array_intersect always returns an array, just an empty one when no results are found. Calling foreach on an empty array won't throw an error or anything, it will simply do nothing.

```suggestion
```

Reply:
This just provides more readable/understandable code IMHO.